examples/example-node-connection-and-error/src/realm-auth.ts

```diff
@@ -0,0 +1,314 @@
+ * time for your App's refresh tokens to be anywhere between 30 minutes and 180 days. See:
+ * https://www.mongodb.com/docs/atlas/app-services/users/sessions/#configure-refresh-token-expiration
+ */
+export async function logIn(email: string, password: string): Promise<boolean> {
+  // If there is already a logged in user, there is no need to reauthenticate.
+  if (currentUser) {
+    return true;
+  }
+
+  try {
+    logger.info("Logging in...");
+    // The credentials here can be substituted using a JWT or another preferred method.
+    currentUser = await app.logIn(Credentials.emailPassword(email, password));
+    originalAccessToken = currentUser.accessToken;
+    logger.info("Logged in.");
+
+    // Listen for changes to user-related events.
+    currentUser.addListener(handleUserEventChange);
+    return true;
+  } catch (err: any) {
+    logger.error(`Error logging in: ${err?.message}`);
+    return false;
+  }
+}
+
+export async function logOut(): Promise<void> {
+  if (currentUser) {
+    logger.info("Logging out...");
+    await currentUser.logOut();
+    // The `currentUser` variable is being set to `null` in the user listener.
+  }
+}
+
+/**
+ * Configure and open the synced realm.
+ */
+export async function openRealm(): Promise<void> {
+  try {
+    if (!currentUser) {
+      throw new Error("The user needs to be logged in before the synced Realm can be opened.");
+    }
+
+    const config: ConfigurationWithSync = {
+      schema: [Store, Kiosk, Product],
+      sync: {
+        user: currentUser,
+        // To read more about flexible sync and subscriptions, see:
+        // https://www.mongodb.com/docs/realm/sdk/node/examples/flexible-sync/
+        flexible: true,
+        initialSubscriptions: {
+          // When adding subscriptions, best practice is to name each subscription
+          // for better managing removal of them.
+          update: (mutableSubs: Realm.App.Sync.MutableSubscriptionSet, realm: Realm) => {
+            // Subscribe to the store with the given ID.
+            mutableSubs.add(
+              realm.objects(Store).filtered("_id = $0", SYNC_STORE_ID),
+              { name: "storeA" },
+            );
+            // Subscribe to all kiosks in the store with the given ID.
+            mutableSubs.add(
+              realm.objects(Kiosk).filtered("storeId = $0", SYNC_STORE_ID),
+              { name: "kiosksInStoreA" },
+            );
+            // Subscribe to all products in the store with the given ID.
+            mutableSubs.add(
+              realm.objects(Product).filtered("storeId = $0", SYNC_STORE_ID),
+              { name: "productsInStoreA" },
+            );
+          },
+        },
+        clientReset: {
+          // Download a fresh copy from the server if recovery of unsynced changes is not possible.
+          mode: ClientResetMode.RecoverOrDiscardUnsyncedChanges,
+          onBefore: handlePreClientReset,
```

Comment:
I think it would be cool to show how to force a client reset.  I suppose that could be done by making a field optional or renaming it.

Reply:
I think we have to take a closer look at how the client reset listeners are fired for our SDK as I tried a similar schema breaking change on the server and only having the sync error handler being called with the `"name":"ClientReset"`. In that case, I can come back and update this app with more examples and information.